src/vos/vos_ts.c

```diff
@@ -204,24 +158,41 @@ vos_ts_table_alloc(struct vos_ts_table **ts_tablep)
 		info->ti_count = type_counts[i];
 		info->ti_table = ts_table;
 		switch (i) {
-		case VOS_TS_TYPE_CONT:
+		case VOS_TS_TYPE_OBJ:
 			miss_size = OBJ_MISS_SIZE;
 			break;
-		case VOS_TS_TYPE_OBJ:
+		case VOS_TS_TYPE_DKEY:
 			miss_size = DKEY_MISS_SIZE;
 			break;
-		case VOS_TS_TYPE_DKEY:
+		case VOS_TS_TYPE_AKEY:
 			miss_size = AKEY_MISS_SIZE;
 			break;
-		case VOS_TS_TYPE_AKEY:
+		case VOS_TS_TYPE_CONT:
 		default:
 			miss_size = 0;
 			break;
 		}
 		if (miss_size) {
 			info->ti_cache_mask = miss_size - 1;
 			info->ti_misses = miss_cursor;
-			miss_cursor += info->ti_count * miss_size;
+			miss_cursor += miss_size;
+			/* Negative entries are global.  Each object/key chain
+			 * will hash to an index.   There will be some false
+			 * sharing but it should be fairly minimal.  Start each
+			 * negative entry with global settings.
+			 */
+			for (j = 0; j < miss_size; j++) {
+				entry = &info->ti_misses[j];
+				entry->te_info = info;
+				vos_ts_copy(&entry->te_ts.tp_ts_rl,
```

Comment:
Can we just keep these entries as "zero" status until they are used for the first time?

Reply:
I suppose we could but this is a one time expense at startup and a bit simpler.   If I do it later, I need to check on every usage.   I'm indifferent on this.

Reply:
Even if we copied from ts_table here, when we use related entry, we still need compare, right? What is the difference between comparing with the ts_table initial value and comparing with zero?

Reply:
I'm not sure what you mean.     The initial value is the startup timestamp and ensures that nothing can be written at an earlier time.   Initializing the negative entries here makes it so we don't need to copy the intial timestamps on lookup.  Also the global timestamps can change due to evictions but the negative timestamps are never evicted so no reason to get updated timestamps at runtime, I think